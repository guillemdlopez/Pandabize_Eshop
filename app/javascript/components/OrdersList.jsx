import React from "react";
import OrderCard from "./OrderCard";
import PropTypes from "prop-types";

const OrdersList = ({ orders }) => {
  return (
    <>
      {orders?.reverse().map((order) => {
        return <OrderCard {...order} key={order.id} />;
      })}
    </>
  );
};

OrdersList.propTypes = {
  orders: PropTypes.array,
};

export default OrdersList;
