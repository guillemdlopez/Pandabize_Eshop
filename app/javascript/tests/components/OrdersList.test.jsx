import { shallow } from "enzyme";
import React from "react";
import OrdersList from "../../components/OrdersList";

describe("Testing the <OrderList /> component", () => {
  test("should be empty when there are no orders", () => {
    const wrapper = shallow(<OrdersList />);
    expect(wrapper.children().length).toBe(0);
  });

  test("should render the component correctly", () => {
    const orders = [
      {
        id: 19,
        bicycle_id: 26,
        amount: 13.1,
        user_id: 10,
        bicycle: "New School XC",
        photo:
          "http://res.cloudinary.com/doker55/image/upload/jbwjj6i6r1f2oz551vxojkmtb42c.jpg",
        customizations: [
          {
            id: 148,
            name: "Wheel size",
            value: "16 inches",
            price: 6.15,
            bicycle_id: 26,
          },
          {
            id: 168,
            name: "Rim color",
            value: "Blue",
            price: 6.95,
            bicycle_id: 26,
          },
        ],
      },
      {
        id: 20,
        bicycle_id: 28,
        amount: 14.26,
        user_id: 10,
        bicycle: "Fast and Furious 5000",
        photo:
          "http://res.cloudinary.com/doker55/image/upload/fj7x3ykjgk56nxonx42wqd67l3il.jpg",
        customizations: [
          {
            id: 154,
            name: "Wheel size",
            value: "15 inches",
            price: 6.28,
            bicycle_id: 28,
          },
          {
            id: 196,
            name: "Saddle color",
            value: "Blue",
            price: 7.98,
            bicycle_id: 28,
          },
        ],
      },
    ];
    const wrapper = shallow(<OrdersList orders={orders} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.children().length).toBe(2);
  });
});
