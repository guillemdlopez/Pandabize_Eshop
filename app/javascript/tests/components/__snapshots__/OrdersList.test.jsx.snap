// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the <OrderList /> component should render the component correctly 1`] = `
<Fragment>
  <OrderCard
    amount={14.26}
    bicycle="Fast and Furious 5000"
    bicycle_id={28}
    customizations={
      Array [
        Object {
          "bicycle_id": 28,
          "id": 154,
          "name": "Wheel size",
          "price": 6.28,
          "value": "15 inches",
        },
        Object {
          "bicycle_id": 28,
          "id": 196,
          "name": "Saddle color",
          "price": 7.98,
          "value": "Blue",
        },
      ]
    }
    id={20}
    key="20"
    photo="http://res.cloudinary.com/doker55/image/upload/fj7x3ykjgk56nxonx42wqd67l3il.jpg"
    user_id={10}
  />
  <OrderCard
    amount={13.1}
    bicycle="New School XC"
    bicycle_id={26}
    customizations={
      Array [
        Object {
          "bicycle_id": 26,
          "id": 148,
          "name": "Wheel size",
          "price": 6.15,
          "value": "16 inches",
        },
        Object {
          "bicycle_id": 26,
          "id": 168,
          "name": "Rim color",
          "price": 6.95,
          "value": "Blue",
        },
      ]
    }
    id={19}
    key="19"
    photo="http://res.cloudinary.com/doker55/image/upload/jbwjj6i6r1f2oz551vxojkmtb42c.jpg"
    user_id={10}
  />
</Fragment>
`;
